metricflow/sql_clients/duckdb.py

```diff
@@ -114,3 +114,7 @@ def create_table_from_dataframe(  # noqa: D
 
     def cancel_request(self, pattern_tag_set: SqlRequestTagSet) -> int:  # noqa: D
         raise NotImplementedError
+
+    def list_tables(self, schema_name: str) -> Sequence[str]:  # noqa: D
+        with self._concurrency_lock:
+            return self._engine.table_names(schema=schema_name)
```

Comment:
I'm assuming this has been tested.

table_names is deprecated so at some point we should add better integration test coverage for this method and move on to the new approach.

Reply:
It was tested as a part of an integration test, and I assumed that a test for `list_tables` was in `test_sql_client.py`. Surprisingly, it was not, so I added one there.

Reply:
I did look into using the `information_schema` for `list_tables`, but there was one issue with Snowflake that I couldn't resolve, so that's been on the back burner.

Reply:
Yeah I tried to update a while back and hit issues as well, probably the same deal. Having the explicit test will hopefully make it easier to futz with it - I'll open an issue.